src/components/tl/ucp/tl_ucp_context.c

```diff
@@ -126,7 +162,23 @@ UCC_CLASS_INIT_FUNC(ucc_tl_ucp_context_t,
 
 UCC_CLASS_CLEANUP_FUNC(ucc_tl_ucp_context_t)
 {
+    ucc_status_t status;
     tl_info(self->super.super.lib, "finalizing tl context: %p", self);
+    if (self->eps) {
+        status = ucc_tl_ucp_close_eps(self, self->eps, self->size);
+        while (UCC_OK != status) {
+            if (status < 0) {
+                tl_error(self->super.super.lib,
+                         "failure during tl_ucp_close_eps, %s",
+                         ucc_status_string(status));
+                break;
+            }
+        }
+        ucc_free(self->eps);
+    }
+    if (self->addr_storage) {
+        ucc_tl_ucp_addr_storage_free(self->addr_storage);
```

Comment:
Maybe move if (self->addr_storage) inside ucc_tl_ucp_addr_storage_free, we have identical checks before it here and in team cleanup